apps/api-gateway/src/core/celery_tasks.py

```
@@ -1354,3 +1354,173 @@ def _write_xlsx(file_path: str, headers: list, rows: list) -> int:
+        total = len(target_tables)
+        row_counts: Dict[str, int] = {}
+        tmp_dir = tempfile.mkdtemp(dir=backup_dir)
+
+        try:
+            for idx, table in enumerate(target_tables):
+                async with async_session() as session:
+                    if backup_type == "incremental" and since_ts:
+                        # 增量：只取 updated_at > since_timestamp 的行
+                        try:
+                            res = await session.execute(
+                                text(f"SELECT * FROM {table} WHERE updated_at > :ts"),
```

Ruff (0.15.2)

[error] Possible SQL injection vector through string-based query construction

(S608)